util/codec/codec.go

```diff
@@ -724,7 +724,7 @@ func Decode(b []byte, size int) ([]types.Datum, error) {
 
 // DecodeRange decodes the range values from a byte slice that generated by EncodeKey.
 // It handles some special values like `MinNotNull` and `MaxValueDatum`.
-func DecodeRange(b []byte, size int) ([]types.Datum, []byte, error) {
+func DecodeRange(loc *time.Location, b []byte, size int, idxColumnType []byte) ([]types.Datum, []byte, error) {
```

Comment:
Please move the argument `loc` to the end and add some comments for it to indicate that it can be nil and only used when the corresponding type is `timestamp`.